coffee/shepherd.coffee

```diff
@@ -117,6 +117,9 @@ class Step extends Evented
     @tether = new Tether extend(tetherOpts, @options.tetherOptions)
 
   show: =>
+      @options.beforeShowPromise?()?.then(=> @_show()) ? @_show()
```

Comment:
I can create a jsfiddle or something, if you need me to reproduce the error for you, but it would be awesome if we could just merge it with the arrow :).